### PR TITLE
Add pass to propagate the layout format

### PR DESCRIPTION
**Pull request by nlpng**
## Motivation and Context

We will use the format of `C`, `WC`, `HWC`, `NHWC` to define the data format.
Therefore need to propagate the format properly to match the runtime. 

## Description

Adding an optimization pass for propagating the layout format

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by iizukak** [comment]
run test

**Comment by iizukak** [comment]
run test

**Comment by nlpng** [comment]
@primenumber @antonionevado 
Could you check whether this serve the requirement of the layout format?

**Comment by primenumber** [comment]
I checked new algorithm correctly infer tensor format in my TensorView codes.

File: dlk/python/dlk/plugins/tf.py [hunk 1]
```diff
@@ -474,62 +474,65 @@ def add_all_nodes(self, graph: Graph) -> None:
...
+        https://www.tensorflow.org/api_docs/python/tf/nn/conv2d
+        - 'QTZ_binary_mean_scaling', 'QTZ_binary_channel_wise_mean_scaling':
+        kernel quantizer is also in HWCN
+        - 'Transpose': depending on the permutation attribute
+        """
+
+        _default_format = 'NHWC'
+        _default_w_format = 'HWCN'
+
+        rank_to_format = {1: 'C', 2: 'WC', 3: 'HWC', 4: 'NHWC'}
```
[the remark this answers is not in the document]
Comment (primenumber) [reply dlk/python/dlk/plugins/tf.py]: We ignore MatMul for now.

**Comment by iizukak** [comment]
@nlpng 
Thank you for resolve confilict 🙇 
Can you remove [WIP] now?

**Pull request merged**